godot-macros/src/docs.rs

```diff
@@ -135,13 +136,66 @@ fn xml_escape(value: String) -> String {
     result
 }
 
-/// Calls [`extract_docs_from_attributes`] and converts the result to BBCode for Godot's consumption.
-fn attribute_docs_to_bbcode(doc: &[venial::Attribute]) -> Option<String> {
-    let doc = extract_docs_from_attributes(doc)
-        .collect::<Vec<String>>()
-        .join("\n");
+/// Returns user docs with its `@deprecated` and `@experimental` attributes extracted.
+fn docs_with_attributes(doc: &[venial::Attribute]) -> (Vec<String>, Vec<String>, Vec<String>) {
+    let (mut docs, mut deprecated, mut experimental): (Vec<String>, Vec<String>, Vec<String>) =
+        (Vec::new(), Vec::new(), Vec::new());
 
-    (!doc.is_empty()).then(|| markdown_converter::to_bbcode(&doc))
+    let docs_ptr = &docs as *const _;
+    let mut current_bucket: &mut Vec<String> = &mut docs;
+
+    for mut line in extract_docs_from_attributes(doc) {
+        // Following lines are being part of given attribute as long as they are not followed by the newline.
+        if !std::ptr::eq(current_bucket, docs_ptr) && line.is_empty() {
+            current_bucket = &mut docs;
+            // Avoid doubling newlines.
+            continue;
+        }
+
+        // Checks if line starts with given attribute tag.
+        let trimmed = line.trim_start();
+        if trimmed.starts_with("@deprecated") {
+            current_bucket = &mut deprecated;
+            line = trimmed.trim_start_matches("@deprecated").to_string();
+        } else if trimmed.starts_with("@experimental") {
+            current_bucket = &mut experimental;
+            line = trimmed.trim_start_matches("@experimental").to_string();
+        }
+
+        current_bucket.push(line);
+    }
+
+    (docs, deprecated, experimental)
+}
+
+/// Converts attribute docs to form suitable for Godot's consumption.
+///
+/// All doc pieces are being converted to BBCode and xml escaped.
+/// Returns user docs with `@deprecated` and `@experimental` attributes or None if no docs are present.
+fn attribute_docs_to_bbcode(doc: &[venial::Attribute]) -> Option<(String, String, String)> {
```

Comment:
From neither the name nor the docs, it's clear that the experimental/deprecated parts XML _attributes_, as in 
```xml
<some_node deprecated="..." experimental="...">
```

This should definitely be mentioned, and also here, maybe a struct would make it clearer:
```rs
struct XmlParagraphs {
    /// XML content, as BBCode: `<description>VALUE</description>`. 
    description_content,

    /// XML attribute, as BBCode: `experimental="VALUE"`. 
    experimental_attr,

    /// XML attribute, as BBCode: `deprecated="VALUE"`. 
    deprecated_attr,
}
```

So experimental/deprecated texts can also be BBCode, not just plain text?

Reply:
Much better :ok_hand: